paddle/framework/variable.h

```diff
@@ -23,12 +23,19 @@ namespace framework {
 
 class Variable {
  public:
+  /**
+   * @brief Get variable holding element.
+   * @tparam T element type
+   * @return nullptr if the variable holds nothing or holder's type and type T
+   * are mismatch, otherwise, return the element pointer.
+   */
   template <typename T>
-  const T& Get() const {
-    PADDLE_ASSERT(holder_ != nullptr);
```

Comment:
I think that here it is alright to use ASSERT here, because the semantic/protocol of Tensor is that programmers use GetMutable to allocate memory, and it is the programmers' responsibility to make sure that Get is called after GetMutable.